### PR TITLE
rafttest: separate network interface and network

### PR DESCRIPTION
**Pull request by xiang90**
@yichengq @bdarnell

### CODE REVIEW

File: raft/rafttest/network.go [hunk 1]
```diff
@@ -7,17 +7,25 @@ import (
...
+// a network
+type network interface {
 	// drop message at given rate (1.0 drops all messages)
 	drop(from, to uint64, rate float64)
 	// delay message for (0, d] randomly at given rate (1.0 delay all messages)
 	// do we need rate here?
 	delay(from, to uint64, d time.Duration, rate float64)
-
 	disconnect(id uint64)
 	connect(id uint64)
```
Comment (yichengq) [inline raft/rafttest/network.go]: are these two functions still necessary?

Comment (xiang90) [inline raft/rafttest/network.go]: yes. why not

Comment (xiang90) [inline raft/rafttest/network.go]: an network interface has at least two ends. one is on your local computer, another one is on the end router.

Comment (yichengq) [inline raft/rafttest/network.go]: k

**Comment by yichengq** [comment]
lgtm

**Pull request merged**